src/base/kaldi-error.cc

```diff
@@ -81,14 +83,14 @@ inline void KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(std::string &ans,
   // to demangle, so we don't check it.
 
   if (demangled_name != NULL) {
-    ans += demangled_name;
+    ans->append(demangled_name);
     free(demangled_name);
   } else {
-    ans += to_append;  // add the original string.
+    ans->append(to_append);  // add the original string.
   }
 }
 #else  // defined(HAVE_CXXABI_H) && defined(HAVE_EXECINFO_H)
-#define KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(ans, to_append) ans += to_append
+#define KALDI_APPEND_POSSIBLY_DEMANGLED_STRING(to_append, ans) ans += to_append
```

Comment:
This is not right because ans is now a pointer... it would just be incrementing the pointer.  You should do *ans += to_append or (better) ans->append(to_append)